### PR TITLE
Embed Preview

### PR DESCRIPTION
**Pull request by adpirz**
Scrolls in mobile, but lets you preview the embed beforehand

image
image

From an embed itself:

image

### CODE REVIEW

File: src/utils/hooks.js [hunk 1]
```diff
@@ -15,17 +22,33 @@ export function useEmbed() {
...
     'frameBorder="0" ' +
     'scrolling="no"' +
     '></iframe>';
 
-  return { isEmbed, iFrameCodeSnippet };
+  const jsCodeSnippet =
+    '<div ' +
+    'class="covid-act-now-embed" ' +
+    (id ? `data-state-id="${id}" ` : '') +
+    (countyId ? `data-fips-id="${countyId}" ` : '') +
```
Comment (adpirz) [inline src/utils/hooks.js:42]: This is not accurate, but how our `embed.js` script works currently (we actually pass in county name strings) -- I can make that fix here though

Comment (mikelehen) [reply src/utils/hooks.js:42]: I think Mike wants us to use the FIPS id in both of these snippets, if possible.

File: src/screens/ModelPage/ShareModelBlock/EmbedPreview.js [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+export default function EmbedPreview({ open, onClose }) {
+  const {
+    EMBED_HEIGHT,
+    EMBED_WIDTH,
+    iFramePath,
+    jsCodeSnippet,
+    iFrameCodeSnippet,
+  } = useEmbed();
+
+  // https://material-ui.com/components/snackbars/#consecutive-snackbars
```
Comment (mikelehen) [inline src/screens/ModelPage/ShareModelBlock/EmbedPreview.js:32]: Perhaps expand this comment to something like:
```
// When you click both copy links back to back, it'll trigger multiple snackbar
// messages.  We show them serially using a queue per example at
// https://material-ui.com/components/snackbars/#consecutive-snackbars
```

File: src/screens/ModelPage/ShareModelBlock/EmbedPreview.js [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+    if (snackbarOpen) {
+      // immediately begin dismissing current message
+      // to start showing new one
+      setSnackbarOpen(false);
+    } else {
+      processQueue();
+    }
+  };
+
+  const handleClose = (event, reason) => {
```
Comment (mikelehen) [inline src/screens/ModelPage/ShareModelBlock/EmbedPreview.js:61]: consider renaming these `handleSnackbarClose` and `handleSnackbarExited` to make it clearer they're for the snackbar, not the modal.

File: src/screens/ModelPage/ShareModelBlock/EmbedPreview.js [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+  } = useEmbed();
+
+  // https://material-ui.com/components/snackbars/#consecutive-snackbars
+  const queueRef = useRef([]);
+  const [snackbarOpen, setSnackbarOpen] = useState(false);
+  const [messageInfo, setMessageInfo] = useState(undefined);
+  const theme = useTheme();
+  const isMobile = useMediaQuery(theme.breakpoints.down('sm'));
+
+  const processQueue = () => {
```
Comment (mikelehen) [inline src/screens/ModelPage/ShareModelBlock/EmbedPreview.js:39]: Consider renaming `queue` and `processQueue` to `snackbarQueue` and `processSnackbarQueue`

File: src/utils/hooks.js [hunk 5]
```diff
@@ -1,31 +1,52 @@
+import { useState, useRef } from 'react';
```
Comment (mikelehen) [inline src/utils/hooks.js:1]: unused

**Pull request merged**